diff --git a/src/com/andrew/apollo/MusicPlaybackService.java b/src/com/andrew/apollo/MusicPlaybackService.java
index 7c34686..eafd83d 100644
--- a/src/com/andrew/apollo/MusicPlaybackService.java
+++ b/src/com/andrew/apollo/MusicPlaybackService.java
@@ -624,7 +624,7 @@ public class MusicPlaybackService extends Service {
                 | RemoteControlClient.FLAG_KEY_MEDIA_PAUSE
                 | RemoteControlClient.FLAG_KEY_MEDIA_PLAY_PAUSE
                 | RemoteControlClient.FLAG_KEY_MEDIA_STOP;
-
+/* still with API 16 
         if (ApolloUtils.hasJellyBeanMR2()) {
             flags |= RemoteControlClient.FLAG_KEY_MEDIA_POSITION_UPDATE;
 
@@ -643,7 +643,7 @@ public class MusicPlaybackService extends Service {
                 }
             });
         }
-
+*/
         mRemoteControlClient.setTransportControlFlags(flags);
     }
 
@@ -1350,10 +1350,13 @@ public class MusicPlaybackService extends Service {
                 ? RemoteControlClient.PLAYSTATE_PLAYING
                 : RemoteControlClient.PLAYSTATE_PAUSED;
 
+	/* still with API 16
         if (ApolloUtils.hasJellyBeanMR2()
                 && (what.equals(PLAYSTATE_CHANGED) || what.equals(POSITION_CHANGED))) {
             mRemoteControlClient.setPlaybackState(playState, position(), 1.0f);
-        } else if (what.equals(PLAYSTATE_CHANGED)) {
+        } else 
+	*/
+	if (what.equals(PLAYSTATE_CHANGED)) {
             mRemoteControlClient.setPlaybackState(playState);
         } else if (what.equals(META_CHANGED) || what.equals(QUEUE_CHANGED)) {
             Bitmap albumArt = getAlbumArt();
@@ -1377,9 +1380,11 @@ public class MusicPlaybackService extends Service {
                     .putBitmap(RemoteControlClient.MetadataEditor.BITMAP_KEY_ARTWORK, albumArt)
                     .apply();
 
+	    /* still with API 16
             if (ApolloUtils.hasJellyBeanMR2()) {
                 mRemoteControlClient.setPlaybackState(playState, position(), 1.0f);
             }
+	    */
         }
     }
 
