diff --git a/src/com/android/settings/cmstats/ReportingServiceManager.java b/src/com/android/settings/cmstats/ReportingServiceManager.java
index 3a065ff4e..eacdb9c3e 100644
--- a/src/com/android/settings/cmstats/ReportingServiceManager.java
+++ b/src/com/android/settings/cmstats/ReportingServiceManager.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2012 The CyanogenMod Project
+ * Copyright (C) 2017 The LineageOS Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -19,7 +20,6 @@ package com.android.settings.cmstats;
 import android.app.AlarmManager;
 import android.app.PendingIntent;
 import android.content.BroadcastReceiver;
-import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
 import android.content.SharedPreferences;
@@ -27,10 +27,14 @@ import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
 import android.util.Log;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+
+
 public class ReportingServiceManager extends BroadcastReceiver {
 
-    public static final long dMill = 24L * 60L * 60L * 1000L;
-    public static final long tFrame = 1L * dMill;
+    public static final long tFrame =  6L * 60L * 60L * 1000L;
 
     @Override
     public void onReceive(Context ctx, Intent intent) {
@@ -46,6 +50,7 @@ public class ReportingServiceManager extends BroadcastReceiver {
         prefs.edit().putBoolean(AnonymousStats.ANONYMOUS_ALARM_SET, false).apply();
         boolean optedIn = prefs.getBoolean(AnonymousStats.ANONYMOUS_OPT_IN, true);
         if (!optedIn) {
+            Log.d(ReportingService.TAG, "User did not opt in when setting alarm.");
             return;
         }
         long lastSynced = prefs.getLong(AnonymousStats.ANONYMOUS_LAST_CHECKED, 0);
@@ -55,12 +60,17 @@ public class ReportingServiceManager extends BroadcastReceiver {
             // sending up anonymous stats.
             lastSynced = System.currentTimeMillis();
         }
+
+        SimpleDateFormat f = new SimpleDateFormat();
         long timeLeft = (lastSynced + tFrame) - System.currentTimeMillis();
-        Intent sIntent = new Intent(ConnectivityManager.CONNECTIVITY_ACTION);
-        sIntent.setComponent(new ComponentName(ctx.getPackageName(), ReportingServiceManager.class.getName()));
+        if (timeLeft < 0)
+             timeLeft = 0;
+
+        Intent sIntent = new Intent (ctx, ReportingServiceManager.class);
+        sIntent.setAction(Intent.ACTION_RUN);
         AlarmManager alarmManager = (AlarmManager) ctx.getSystemService(Context.ALARM_SERVICE);
         alarmManager.set(AlarmManager.RTC_WAKEUP, System.currentTimeMillis() + timeLeft, PendingIntent.getBroadcast(ctx, 0, sIntent, 0));
-        Log.d(ReportingService.TAG, "Next sync attempt in : " + timeLeft / dMill + " days");
+        Log.d(ReportingService.TAG, "Next sync attempt at " + f.format (new Date (System.currentTimeMillis() + timeLeft)));
         prefs.edit().putBoolean(AnonymousStats.ANONYMOUS_ALARM_SET, true).apply();
     }
 
@@ -72,20 +82,28 @@ public class ReportingServiceManager extends BroadcastReceiver {
             long lastSynced = prefs.getLong(AnonymousStats.ANONYMOUS_LAST_CHECKED, 0);
             boolean optedIn = prefs.getBoolean(AnonymousStats.ANONYMOUS_OPT_IN, true);
             boolean alarmSet = prefs.getBoolean(AnonymousStats.ANONYMOUS_ALARM_SET, false);
-            if (alarmSet) {
+            boolean shouldSync = false;
+
+            // simplest case
+            if (!optedIn) {
+                Log.d(ReportingService.TAG, "User did not opt in.");
                 return;
             }
-            boolean shouldSync = false;
-            if (lastSynced == 0) {
-                shouldSync = true;
-            } else if (System.currentTimeMillis() - lastSynced >= tFrame) {
+            // alarm is set and it is in the future
+            if (alarmSet && (System.currentTimeMillis() - lastSynced < tFrame)) {
+                Log.d(ReportingService.TAG, "Alarm already set. Bailing out.");
+                return;
+            }
+            if ((lastSynced == 0) || (System.currentTimeMillis() - lastSynced >= tFrame)) { 
                 shouldSync = true;
             }
-            if (shouldSync && optedIn) {
-                Intent sIntent = new Intent();
-                sIntent.setComponent(new ComponentName(ctx.getPackageName(), ReportingService.class.getName()));
+            if (shouldSync) {
+                Intent sIntent = new Intent (ctx, ReportingService.class);
+		sIntent.setAction(Intent.ACTION_RUN);
                 ctx.startService(sIntent);
-            } else if (optedIn) {
+                Log.d(ReportingService.TAG, "ReportingService started.");
+            } else {
+                Log.d(ReportingService.TAG, "New alarm set.");
                 setAlarm(ctx);
             }
         }
